### PR TITLE
feat(ci_visibility): log correlation and submission via EVP proxy and agentless

### PR DESCRIPTION
**Pull request by gnufede**
## Description

Adds two related features to the `ddtrace.testing` pytest plugin:

**Log correlation (`DD_LOGS_INJECTION=true`)**
The ddtrace logging patch is applied during the test session so that every log record carries `dd.trace_id` and `dd.span_id` matching the active test span.

**Log submission**
Log records emitted during tests are forwarded to the Datadog logs intake:

- **Agent mode** (`DD_LOGS_INJECTION=true`): logs are routed through the agent's EVP proxy.
- **Agentless mode** (`DD_AGENTLESS_LOG_SUBMISSION_ENABLED=true` + `DD_CIVISIBILITY_AGENTLESS_ENABLED=true`): logs are posted directly to the Datadog logs intake.

Log submission is suppressed when `_DD_CIVISIBILITY_USE_CI_CONTEXT_PROVIDER=1`: in that mode a separate `CIVisibilityTracer` with its own `CIContextProvider` is used for test spans, independent from the global `ddtrace.tracer`. Since the logging patch reads from the global tracer, it would see no active test span and produce zero trace IDs. If `DD_AGENTLESS_LOG_SUBMISSION_ENABLED` is set without `DD_CIVISIBILITY_AGENTLESS_ENABLED`, a warning is emitted and log submission is disabled.

Modelled on the Ruby implementation: https://github.com/DataDog/datadog-ci-rb/pull/306

## Testing

New test file `tests/testing/internal/pytest/test_pytest_log_correlation.py` with 8 subprocess-based tests covering:

- `DD_LOGS_INJECTION=true` → records carry correct trace/span IDs matching the active test span
- `DD_LOGS_INJECTION=false` → no trace IDs injected
- Agentless flags set → `LogsHandler` installed and forwards events with correct trace IDs
- `DD_LOGS_INJECTION=true` alone (EVP proxy mode) → `LogsHandler` installed and forwards events
- `DD_LOGS_INJECTION=true` + `_DD_CIVISIBILITY_USE_CI_CONTEXT_PROVIDER=1` → `LogsHandler` not installed
- Missing flags → `LogsHandler` not installed
- Root logger level filtering: child logger warnings suppressed when root is set to ERROR

### CODE REVIEW

File: ddtrace/testing/internal/logs.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                "hostname": self._writer.hostname,
+                "message": record.getMessage(),
+                "service": self._writer.service,
+                "status": record.levelname.lower(),
+                "dd.trace_id": getattr(record, "dd.trace_id", LOG_ATTR_VALUE_ZERO),
+                "dd.span_id": getattr(record, "dd.span_id", LOG_ATTR_VALUE_ZERO),
+            }
+            self._writer.put_event(event)
+        except Exception:
+            self.handleError(record)
```
Comment (mabdinur) [inline ddtrace/testing/internal/logs.py]: Do we have test coverage for this error case?

File: ddtrace/testing/internal/pytest/plugin.py [hunk 2]
```diff
@@ -261,6 +290,32 @@ def pytest_sessionstart(self, session: pytest.Session) -> None:
 
         if self.enable_ddtrace_trace_filter:
             install_global_trace_filter(self.manager.writer)
+            try:
+                import ddtrace
```
Comment (mabdinur) [inline ddtrace/testing/internal/pytest/plugin.py]: If the plugin is being used ddtrace should be installed. Do we need to handle this edge case?

File: docs/configuration.rst [hunk 3]
```diff
@@ -790,6 +790,20 @@ Test Visibility
...
+   DD_AGENTLESS_LOG_SUBMISSION_ENABLED:
+     type: Boolean
+     default: False
+
+     description: |
+        When used with ``DD_CIVISIBILITY_AGENTLESS_ENABLED=true``, enables log submission from the
+        ``ddtrace.testing`` pytest plugin directly to the Datadog logs intake. Log records emitted
+        during tests are enriched with the active test span's ``dd.trace_id`` and ``dd.span_id``
+        and forwarded as structured JSON. When using the Datadog Agent instead of agentless mode,
+        set ``DD_LOGS_INJECTION=true`` to achieve the same result via the agent's EVP proxy.
```
Comment (KowalskiThomas) [inline docs/configuration.rst]: Is `EVP` a public-facing term/something we already have in the docs? If not it may be worth changing to something more generic like "intake" or something.

Comment (gnufede) [reply docs/configuration.rst]: True that it is not very well documented, just mentioned, for example here: https://docs.datadoghq.com/agent/configuration/dual-shipping/?tab=helm#yaml-configuration

Comment (gnufede) [inline docs/configuration.rst]:
```suggestion
        set ``DD_LOGS_INJECTION=true`` to achieve the same result via the agent's intake.
```
